### DISCUSSION TITLE
Guide: Controlling Airjets with a PWM Speed Controller

### DISCUSSION THREAD

**Discussion by Addonnis** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#412] [Show and tell]
The airjets are controlled by a simple motor. By adjusting the motor speed, you can control the amount of air it pushes, and subsequently the strength of the bubbles in your spa.

This has several benefits:
- Less power consumption
- Quieter jets (you can talk at a normal volume over the bubbles!)
- Better heat retention; when running bubbles on low, the heater is able to maintain a consistent temperature. Very helpful for extended sessions without the water temp dropping.

Usual warnings:
- Disconnect pump from power completely before performing any work.
- Don't attempt if you're not comfortable with electrics, power etc.
- Am sure it voids any warranty.

Required:
- Screwdriver to dissemble the pump
- Power drill
- 6mm drill bit
- x3 small lengths (~15-20cm) of decent gauge wiring
- x6 spade/fork terminals for your new wiring (optional, but recommended)
- A PWM controller like this: 
![image](https://user-images.githubusercontent.com/97511696/206883357-57a06f29-2a83-4d77-aa3f-5c4216339c59.png)
Search eBay for: **2000W AC 50-220V 25A PWM VOLTAGE REGULATOR ADJUSTABLE MOTOR SPEED CONTROLLER**

The controller basically 'splices in' between the airjet pump and the power supply. To install:

1. Remove the top case of the pump. 
2. Remove the black cover to the power block/logic board.
3. Note where the airjet connects on the spa power block. There are only two wires running from the motor, these are easy to identify. Take a photo before proceeding if you're not confident.
4. Disconnect the airjet terminals from the power block, and fit x2 new wires in their place.
5. Connect the airjet wiring to the "outputs" on your PWM controller
6. Connect your new wiring from the power block to the "inputs" on your PWM controller:
7. Optional: Connect a third wire from the PWM controller to the pump ground. I recommend this (I used one of the screws on the board).
![image](https://user-images.githubusercontent.com/97511696/206883990-adeb497f-15b6-4f14-8e96-4b5729f4fd02.png)
9. Power on the spa and test the bubbles - Confirm they're blowing, and they're adjustable with the controller knob.
10. If you're happy everything is working, you'll now need to mount the controller. 
11. Find a spot on the side of your pump you're happy with. You'll need to ensure this clears the pump components when the case is reassembled, and also there is enough length of wire to reach where you're mounting. You could extend the wiring if you really wanted, but I didn't.
12. Drill out a hole in the side of your pump case, pop your controller in and tighten the nut (Finger tight + a quarter turn is fine, if you overtighten you can put stress on the PWM board/components)
![image](https://user-images.githubusercontent.com/97511696/206884113-b14fc2bf-04f5-4119-9034-27f8f44ed537.png)
13. Fit your knob, zip up the top case and enjoy your adjustable bubbles.

I have used this same controller on both the egg-shape and square-shape pumps, it works great on both. 

This controller + my ESP8266 = Very happy spa-ing!

Thanks VisualApproach for his hard work on the Wi-Fi controller.

**Comment by Addonnis** [comment]
Example of the water on low and high settings. 

![image](https://user-images.githubusercontent.com/97511696/206884346-8ad41909-accf-4082-9a4a-8efb25e0b012.png)

**Reply by torei** [reply]
wow, that looks really nice. im not using the bubbles at all because they are just too loud.
will try to do this on my pump as well :] thank you

**Reply by Addonnis** [reply]
You got me curious about the actual sound levels, I used an app to test (not scientific I know, but good enough for a gauge)

Idle/no bubbles: 50db
Bubbles on low: 60db
Bubbles on high: 80db

Goodluck with the mod!

**Reply by AttiMM** [reply]
Thanks! I think I will make this modification.
Guys, do you (@Addonnis, @torei, @visualapproach ) think that is it possible to heat the water while hydrojetting? 
I know that pump display/controller automatically turn off heating when I switch to hydrojet - It's logical because the water flowing is too strong for heating. 
But what if I use another PWM controller for hydrojet to reduce water pressure? I would like to have a gentle warm water flow thru the jets while it's heated.
What do you think: is it feasible?

**Reply by visualapproach** [reply]
> Thanks! I think I will make this modification.
> 
> Guys, do you (@Addonnis, @torei, @visualapproach ) think that is it possible to heat the water while hydrojetting? 
> 
> I know that pump display/controller automatically turn off heating when I switch to hydrojet - It's logical because the water flowing is too strong for heating. 
> 
> But what if I use another PWM controller for hydrojet to reduce water pressure? I would like to have a gentle warm water flow thru the jets while it's heated.
> 
> What do you think: is it feasible?

The total power consumption may be more than it's designed for so beware

**Reply by AttiMM** [reply]
How do you mean: "total"? I thought I would use 2 separate PWM controllers: one for the bubbles and another one for water jets. I think none of them has more than 2 kWh power consumption.

But what is the answer to my main question? Could you turn on heating while water jetting with your module - despite it's not possible with the pump display/controller?

(I know you are working on the new firmware - so my question is theoretic at the moment: should I buy 2 PCM controllers?)
Thx

**Reply by visualapproach** [reply]
Hi
I think it's possible to trick the 4-wire pumps, but not the 6-wire.
Regarding the total power I mean the pump with it's power cord and fuses have a max ampere/watts rating. Say 2kW. If you then force 3-4 kW you're out of spec and your house will catch on fire. Or the fuses will blow. Ofc you can regulate it with the PWM but I'm just saying, so you think about it.

**Reply by AttiMM** [reply]
I think now I understand your thoughts: using waterjet pump and heating at the same time can be too much consumption!
And because I have a 6 wire pump, I think I should let my idea go.

**Reply by torei** [reply]
just finished my mod :]
thanks again @Addonnis 

![grafik](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/6784876/bb794b81-aeb0-4870-bc01-c2a9bfc3c628)

**Reply by Addonnis** [reply]
> just finished my mod :] thanks again @Addonnis
> 
> ![grafik](https://user-images.githubusercontent.com/6784876/242860357-bb794b81-aeb0-4870-bc01-c2a9bfc3c628.png)

Awesome! How do you like it over the standard bubbles?

**Reply by torei** [reply]
i didn't try yet. no water in the pool.
i was cleaning it yesterday, so i used the time to build your idea :]
will probably put an isolation level below the pool before filling it with water again
